### PR TITLE
Update to v0.15.4

### PR DESCRIPTION
**Pull request by sigma-andex**
Update pursuit to 0.15.4. 

Apparently `barrier`, the library to render badges, is hopelessly outdated and doesn't work with GHC 9.2.3. I haven't found any replacement for rendering badges. Not sure if anyone wants to fork, update and maintain this library? Otherwise I would propose to drop badge support.

### CODE REVIEW

**Comment by sigma-andex** [comment]
@PureFunctor could you check if my updates in `src/SearchIndex.hs` are correct?

File: src/SearchIndex.hs
Comment (PureFunctor) [inline src/SearchIndex.hs:300]: Aside from the redundant parentheses, this LGTM

```suggestion
    parentName = P.Qualified P.ByNullSourcePos (P.ProperName declTitle)
```

File: config/routes
Comment (PureFunctor) [inline config/routes:13]: I'm not actually sure we can drop badge support altogether. If I recall correctly, most of the packages within core make use of this for their READMEs. If we need to fork `badger` to work with 9.2.3 then I could probably pick up that task.

**Review by PureFunctor — CHANGES_REQUESTED** [review]

File: config/routes
Comment (PureFunctor) [reply config/routes:13]: Looks like it just needs its upper bound updated: https://github.com/PureFunctor/barrier/commit/79e8fc36fb7d36591f9f65c0e0eb892d8062cdc5

Comment (sigma-andex) [reply config/routes:13]: cool, yeah that sounds doable.

Comment (JordanMartinez) [reply config/routes:13]: Looks like Phil did the same thing a few years ago: https://github.com/philopon/barrier/pull/4

Comment (sigma-andex) [reply config/routes:13]: ok I guess we can try open a PR and see if we get a response

Comment (sigma-andex) [reply config/routes]: Ok I undid the badge removal and added @PureFunctor 's fork/commit as a local dependency. This builds for me using cabal. I'm not sure if it works on stack, as I cannot get stack to run. Would be great if one of you could test it.
Or do we actually need/use stack?

Comment (PureFunctor) [reply config/routes]: Yeah, we need to specify a git dependency for it under `extra-deps`. I don't have a `stack` built at hand at the moment, since it's pretty wonky at M1. I will transfer over my fork to the `purescript` organization in the meantime as well.

Comment (sigma-andex) [reply config/routes]: I added it to extra-deps, but I can't get it to build.

Comment (PureFunctor) [reply config/routes]: Looking into it...

Comment (PureFunctor) [reply config/routes]: Ok, I think I need to relax the bounds for `bytestring` in `barrier`: PureFunctor/barrier@db5e27c

Comment (sigma-andex) [reply config/routes]: Thanks to @i-am-the-slime I got the stack build working as well.

Comment (sigma-andex) [reply config/routes]: @PureFunctor if you move it to `purescript-contrib` I can update the dependencies. Apart fromt his I think it is good to go now

File: src/Handler/Caching.hs
Comment (JordanMartinez) [inline src/Handler/Caching.hs:105]: Should this be readded?

File: stack.yaml
Comment (JordanMartinez) [inline stack.yaml:31]: I'm not familiar with `stack.yaml` in general. Why are these needed?

File: src/Handler/Caching.hs
Comment (sigma-andex) [reply src/Handler/Caching.hs:105]: uh yes

File: stack.yaml
Comment (sigma-andex) [reply stack.yaml:31]: oh yes that accidenily slipped into the commit

File: src/Handler/Caching.hs
Comment (sigma-andex) [reply src/Handler/Caching.hs]: Fixed

File: stack.yaml
Comment (sigma-andex) [reply stack.yaml]: Fixed

**Review by JordanMartinez — APPROVED** [review]

**Comment by PureFunctor** [comment]
Woops, CI breaks because I force pushed on my `barrier` fork 😅. Updating the commit hash to PureFunctor/barrier@db5e27c should make this work.

**Comment by sigma-andex** [comment]
Ok updated it to this commit

**Comment by sigma-andex** [comment]
@JordanMartinez can you re-run the pipeline?

**Comment by JordanMartinez** [comment]
> @JordanMartinez can you re-run the pipeline?

Restarted. Sorry about the delay.

**Comment by sigma-andex** [comment]
Next try :D

**Comment by sigma-andex** [comment]
Ok I think now it should be good :D

**Comment by sigma-andex** [comment]
Once this is good, it would be great to get #469 approved & merged as well and then do a deployment

**Comment by JordanMartinez** [comment]
@garyb Assuming this builds, can I get another approval here? I'm not sure if @PureFunctor's approval (if it was given) would count due to not being a core team member. If I'm wrong, feel free to correct me.

**Comment by sigma-andex** [comment]
> @garyb Assuming this builds, can I get another approval here? I'm not sure if @PureFunctor's approval (if it was given) would count due to not being a core team member. If I'm wrong, feel free to correct me.

How about promoting @PureFunctor to the core team so you can be sure about it? :)

**Review by JordanMartinez — APPROVED** [review]

**Comment by JordanMartinez** [comment]
Thanks all!

**Pull request merged**
